File: src/main/resources/scalagen/finagleService.scala
// ----- finagle service

import com.twitter.scrooge.FinagleThriftService

class FinagledService(
  iface: FutureIface,
  {{override}}val protocolFactory: TProtocolFactory
) extends {{extends}} {
{{#functions}}
{{function}}
{{/function}}
}

File: src/main/resources/javagen/finagleService.java
// ----- finagle service

import com.twitter.scrooge.FinagleThriftService;

class FinagledService extends {{parent}} {
  private FutureIface iface;
  private TProtocolFactory protocolFactory;

  public FinagledService(FutureIface iface, TProtocolFactory protocolFactory) {
{{#hasParent}}
    super(iface, protocolFactory);
{{/hasParent}}
    this.iface = iface;
    this.protocolFactory = protocolFactory;

{{#functions}}
  {{>function}}
{{/function}}
  }
}
